src/cargo/ops/cargo_compile/mod.rs

```diff
@@ -657,8 +663,13 @@ fn traverse_and_share(
         _ => unit.kind,
     };
 
+    let cfg = target_data.cfg(unit.kind);
+    let is_target_windows_msvc = cfg.contains(&Cfg::Name("windows".to_string()))
+        && cfg.contains(&Cfg::KeyPair("target_env".to_string(), "msvc".to_string()));
     let mut profile = unit.profile.clone();
-    if profile.strip.is_deferred() {
+    // For MSVC, rustc currently treats -Cstrip=debuginfo same as -Cstrip=symbols, which causes
+    // this optimization to also remove symbols and thus break backtraces.
+    if profile.strip.is_deferred() && !is_target_windows_msvc {
```

Comment:
Should we have a simple test for this on Windows MSVC to ensure the patch really fixes it?

Reply:
We already have a test that this applies on Windows, since CI fails :) I can add another one that checks that backtraces seem to contain some actual symbols. Is there a preferred way of writing platform specific tests?

Reply:
It might take some fiddling but you can reference this test-crate for the test that should've worked with *some* reasonable configuration of the Cargo.toml: https://github.com/rust-lang/backtrace-rs/tree/6fa4b85b9962c3e1be8c2e5cc605cd078134152b/crates/without_debuginfo